tools/COMPILE-PROTOS.sh

```diff
@@ -82,7 +82,15 @@ END
 
 # patch the generated GAPIC to send Authorization tokens with insecure requests
 # (this allows the registry command line tools to test container builds)
-sed -i.bak 's/return metadata.NewOutgoingContext(ctx, out)/insecure := os.Getenv("APG_REGISTRY_INSECURE")\ntoken := os.Getenv("APG_REGISTRY_TOKEN")\nif insecure == "1" \&\& token != "" \{ \nout["authorization"] = append(out["authorization"], "Bearer "+token) \n\}\nreturn metadata.NewOutgoingContext(ctx, out)/' gapic/doc.go
+PATCH='\
+insecure := os.Getenv("APG_REGISTRY_INSECURE")\
+token := os.Getenv("APG_REGISTRY_TOKEN")\
+if insecure == "1" \&\& token != "" {\
+  out["authorization"] = append(out["authorization"], "Bearer "+token)\
+}\
+return metadata.NewOutgoingContext(ctx, out)\
+' # no ' or / in the patch; escape & and newlines
+sed -i.bak "s/return metadata.NewOutgoingContext(ctx, out)/${PATCH}/" gapic/doc.go
```

Comment:
Did you check this on Linux? We're not checking this in CI but if we did, that would clarify the safety of this - I think a simple grep for "APG_REGISTRY_INSECURE" in the generated file would verify that the patch was applied.

Reply:
Could you add this to COMPILE-PROTOS.sh right after the `gofmt` call?
```
grep -q APG_REGISTRY_INSECURE gapic/doc.go
if [ $? -ne 0 ]
then
  echo "GAPIC patch failed"
  exit -1
fi
```

This will cause the make to fail, which will show up in CI.

Reply:
Also tested on Linux. Added grep check.